src/sentry/newsletter/base.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import absolute_import
+
+
+class Newsletter(object):
+    PRODUCT_CAMPAIGN = 1
```

Comment:
I would probably not call it Newsletter if its intended to be the solution for general subscriptions.

Basically you have X "campaigns" (e.g. product newsletter) they can subscribe to, and they can come from Y external campaigns (e.g. source=blog). Hypothetically campaigns could be as simple as like "conference updates" which are one-off things, so it makes the terminology a bit confusing with Newsletter.

I'm not sure if each campaign requires opt-in (I dont think it does), but rather they need verified once, and then the ability to opt-in/out of each campaign via settings.